src/layouts/index.jsx

```diff
@@ -4,33 +4,53 @@ import GlobalStore from '../state/GlobalStore';
 // import '../assets/stylesheets/styles.scss';
 import styles from './layout.module.scss';
 import Header from '../components/site/header';
-// import Footer from '../components/site/footer';
+import TableOfContents from '../components/site/tableOfContents';
 
 class Layout extends React.Component {
   constructor() {
     super();
+
     const store = new GlobalStore();
 
     store.addCallbacks();
     store.addReducers();
+
+    this.state = {
+      openSidebar: false,
+    };
   }
 
+  toggleSidebar = () => {
+    const { openSidebar } = this.state;
+    this.setState({ openSidebar: !openSidebar });
```

Comment:
Please replace with a future-proofed setState that incorporates whatever parts of state object we're not touching:
```
this.setState(prevState => ({
   ...prevState,
   openSidebar: !openSidebar,
}));
```
that way if this state object gets more complicated in the future this method won't bulldoze what's there.

Reply:
I like this. Need to adapt my development style to the React way of life